extensions/eclipse-che-theia-plugin-ext/src/browser/oauth-utils.ts

```diff
@@ -65,16 +65,34 @@ export class OauthUtils {
     }
 
     async getToken(oAuthProvider: string): Promise<string | undefined> {
-        return await this.cheApiService.getOAuthToken(oAuthProvider, await this.getUserToken());
+        const token = await this.getUserToken();
+
+        if (token === undefined) {
+            return undefined;
```

Comment:
We can't return `undefined` if token is missing. This can happen In case of the single-user mode. `che-workspace-client` handles this case by itself: https://github.com/eclipse/che-workspace-client/blob/dfaf9f1c543eb5ddb43146c7c1248187328d8030/src/rest/resources.ts#L302-L305

Reply:
che-api-service handles what type of token should we provide for making requests: https://github.com/eclipse/che-theia/blob/80548d47b83a073e71128b80cd51d47009a3f1fd/extensions/eclipse-che-theia-plugin-ext/src/node/che-api-service.ts#L233-L234

And AFAIK, in single-user mode this method should return keycloak token most of the time.

Reply:
And on the other hand, there are bunch of methods in che-api-service, that won't work with machine token, such as `getAll` and `getAllByNamespace`, so, anyway they should have user token. When oauth-utils doesn't provide user token, we'll know, that something happened on obtaining token step.

Reply:
We can throw an Error object here, for example.

Reply:
> And AFAIK, in single-user mode this method should return keycloak token most of the time.

This function `getToken(oAuthProvider: string)` returns the token of the specified oAuth provider (GitHub, OpenShift ...). 
> When oauth-utils doesn't provide user token, we'll know, that something happened on obtaining token step. We can throw an Error object here, for example. Then we'll be confident, that we'll have token anyway.

It should be done in the step of obtaining the user token.

Reply:
So, `getToken(oAuthProvider: string)` signature can be modified to return only `string` value, w/o `undefined` value.

> It should be done in the step of obtaining the user token.

Make sense, I'll update the PR.

Reply:
> So, `getToken(oAuthProvider: string`) signature can be modified to return only `string` value, w/o `undefined` value.

agree

Reply:
Sorry, I forgot, we need `undefined` in case of the single-user

Reply:
Code reverted.